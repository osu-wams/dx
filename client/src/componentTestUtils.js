import React from 'react';
import { render as rtlRender } from 'react-testing-library';
import { createHistory, createMemorySource, LocationProvider } from '@reach/router';
import { ThemeProvider } from 'styled-components';
import theme from './theme';

function render(ui, ...rest) {
  return rtlRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, ...rest);
}

// this is a handy function that I would utilize for any component
// that relies on the router being in context
function renderWithRouter(
  ui,
  { route = '/', history = createHistory(createMemorySource(route)) } = {}
) {
  return {
    ...rtlRender(<LocationProvider history={history}>{ui}</LocationProvider>),
    // adding `history` to the returned utilities to allow us to reference it in our tests
    history
  };
}

export * from 'react-testing-library';
// override the built-in render with our own
export { render, renderWithRouter };
