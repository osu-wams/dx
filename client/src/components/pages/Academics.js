import React, { Component } from 'react';
import Courses from '../Courses';
import PageTitle from '../layout/PageTitle';

export default class Academics extends Component {
  constructor(props) {
    super(props);
    this.state = { api: 'Api' };
  }

  render() {
    const { api } = this.state;
    return (
      <div data-testid="academics-page">
        <PageTitle title="Academics" />
        <Courses />
        <p>Thisis: {api}</p>
      </div>
    );
  }
}
