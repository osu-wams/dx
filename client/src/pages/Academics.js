import React from 'react';
import Courses from '../features/Courses';
import PageTitle from '../ui/PageTitle';
import {
  Overview,
  OverviewContent,
  OverviewSection,
  OverviewItem,
  OverviewItemLabel
} from '../ui/Overview';
import UpcomingAssignments from '../features/UpcomingAssignments';
import ServiceCard from '../features/ServiceCard';
import { DesktopGrid, MainColumn } from '../ui/DesktopGrid';

const AcademicOverview = () => {
  return (
    <Overview>
      <OverviewContent>
        <OverviewSection>
          <OverviewItem style={{ lineHeight: 1 }}>
            <OverviewItemLabel color="stratosphere">Current GPA</OverviewItemLabel>
            <span style={{ fontWeight: 'bold', fontSize: '80px' }}>3.8</span>
          </OverviewItem>
          <OverviewItem>
            <span>
              Week 10
              <br />
              Fall Term
            </span>
          </OverviewItem>
        </OverviewSection>
        <OverviewSection>
          <OverviewItem>
            <OverviewItemLabel color="stratosphere">Major</OverviewItemLabel>
            <span>Graphic Design</span>
          </OverviewItem>
          <OverviewItem>
            <OverviewItemLabel color="stratosphere">Minor</OverviewItemLabel>
            <span>French</span>
          </OverviewItem>
          <OverviewItem>
            <OverviewItemLabel color="stratosphere">Academic Standing</OverviewItemLabel>
            <span>Honor Roll</span>
          </OverviewItem>
        </OverviewSection>
      </OverviewContent>
    </Overview>
  );
};

const Academics = () => {
  return (
    <div data-testid="academics-page">
      <AcademicOverview />
      <PageTitle title="Academics" />
      <DesktopGrid>
        <MainColumn>
          <UpcomingAssignments />
          <Courses />
        </MainColumn>
        <ServiceCard title="Academic Services" serviceType="academic" />
      </DesktopGrid>
    </div>
  );
};

export default Academics;
