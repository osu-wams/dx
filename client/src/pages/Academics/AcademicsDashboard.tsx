import React from 'react';
import Courses from '../../features/Courses';
import AnnouncementContainer from '../../ui/AnnouncementContainer';

const AcademicsDashboard = () => {
  return (
    <>
      <Courses />
      <AnnouncementContainer type="academic" />
    </>
  );
};

export default AcademicsDashboard;
