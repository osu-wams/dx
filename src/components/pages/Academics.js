import React, { Component } from 'react';
import Courses from '../Courses';

export default class Academics extends Component {
  constructor(props) {
    super(props);
    this.state = { api: 'Api' };
  }

  render() {
    const { api } = this.state;
    return (
      <div>
        <Courses />
        <p>Thisis: {api}</p>
      </div>
    );
  }
}
