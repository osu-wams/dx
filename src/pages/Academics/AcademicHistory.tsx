import React, { useState, useEffect } from 'react';
import Skeleton from 'react-loading-skeleton';
import styled from 'styled-components';
import VisuallyHidden from '@reach/visually-hidden';
import { useDebounce } from 'use-debounce';
import { faSearch } from '@fortawesome/pro-light-svg-icons';
import { Grades } from '../../api/student/grades';
import { getGrades } from '../../api/student';
import PlainCard from '../../ui/PlainCard';
import { Color } from '../../theme';
import Input from '../../ui/Input';
import Icon from '../../ui/Icon';
import PageTitle from '../../ui/PageTitle';
import {
  Table,
  TableBody,
  TableRow,
  TableCell,
  TableHeader,
  TableHeaderCell
} from '../../ui/Table';
import { singularPlural, titleCase } from '../../util/helpers';
import { Event } from '../../util/gaTracking';
import { MainGridWrapper, MainGrid, MainGridCol } from '../../ui/PageGrid';
import { AcademicSubNav } from './AcademicsSubNav';

const AcademicHistory = () => {
  const [grades, setGrades] = useState<Grades[]>([]);
  const [query, setQuery] = useState('');
  const [debouncedQuery] = useDebounce(query, 300);
  const [gradesLoading, setGradesLoading] = useState<boolean>(true);
  const [filteredGrades, setFilteredGrades] = useState<Grades[]>([]);

  // Populate user grades
  useEffect(() => {
    let isMounted = true;
    getGrades()
      .then(data => {
        if (isMounted) {
          setGrades(data);
          setFilteredGrades(data);
          setGradesLoading(false);
        }
      })
      .catch(console.log);

    return () => {
      // prevents setting data on a component that has been unmounted before promise resolves
      isMounted = false;
    };
  }, []);

  useEffect(() => {
    if (!debouncedQuery) {
      setFilteredGrades(grades);
    } else {
      const re = new RegExp(debouncedQuery, 'gi');
      const matchingGrades = grades.filter(
        e =>
          e.attributes.courseTitle.match(re) ||
          `${e.attributes.courseSubject}${e.attributes.courseNumber}`.match(re) ||
          `${e.attributes.courseSubject} ${e.attributes.courseNumber}`.match(re)
      );
      setFilteredGrades(matchingGrades);
      Event('academic-history-search', debouncedQuery);
    }
  }, [debouncedQuery, grades]);

  const gradesByTerm = filteredGrades.reduce(
    (gradesSoFar, { attributes, attributes: { termDescription } }) => {
      if (!gradesSoFar[termDescription]) gradesSoFar[termDescription] = [];
      gradesSoFar[termDescription].push(attributes);
      return gradesSoFar;
    },
    {}
  );

  return (
    <MainGridWrapper data-testid="academic-history">
      <PageTitle title="Academic History" />
      <AcademicSubNav />
      <MainGrid>
        <MainGridCol className="col-span-2">
          <SearchWrapper>
            <Icon icon={faSearch} color={Color['neutral-600']} />
            <VisuallyHidden>
              <label htmlFor="course-filter">Find courses</label>
            </VisuallyHidden>
            <FilterInput
              type="text"
              placeholder="Find past courses"
              value={query}
              id="course-filter"
              onChange={e => setQuery(e.target.value)}
            />
          </SearchWrapper>
          {gradesLoading && <Skeleton count={5} />}
          {grades.length > 0 ? (
            <div aria-live="polite" aria-atomic="true">
              {Object.keys(gradesByTerm).map((key, index) => (
                <PlainCard title={key} key={index}>
                  <Table variant="basic" stretch>
                    <TableHeader>
                      <TableRow>
                        <TableHeaderCell>Course Code</TableHeaderCell>
                        <TableHeaderCell>Final Grade</TableHeaderCell>
                        <TableHeaderCell>Course Title</TableHeaderCell>
                        <TableHeaderCell>Credits</TableHeaderCell>
                      </TableRow>
                    </TableHeader>
                    <TableBody>
                      {gradesByTerm[key].map(
                        (
                          { courseNumber, courseSubject, creditHours, gradeFinal, courseTitle },
                          subindex
                        ) => {
                          return (
                            <TableRow key={subindex}>
                              <TableCell>
                                <strong>
                                  {courseSubject} {courseNumber}
                                </strong>
                              </TableCell>
                              <TableCell>{gradeFinal}</TableCell>
                              <TableCell>{titleCase(courseTitle)}</TableCell>
                              <TableCell>
                                {creditHours} {singularPlural(creditHours, 'Credit')}
                              </TableCell>
                            </TableRow>
                          );
                        }
                      )}
                    </TableBody>
                  </Table>
                </PlainCard>
              ))}
            </div>
          ) : (
            !gradesLoading && <div>No course history yet</div>
          )}
        </MainGridCol>
      </MainGrid>
    </MainGridWrapper>
  );
};

export default AcademicHistory;

const SearchWrapper = styled.div`
  position: relative;
  svg {
    position: absolute;
    top: 1rem;
    left: 1.2rem;
  }
  margin-bottom: 2rem;
`;
const FilterInput = styled(Input)`
  width: 100%;
  padding-left: 4rem;
`;
