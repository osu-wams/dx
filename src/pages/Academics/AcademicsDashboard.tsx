import React from 'react';
import { faGraduationCap } from '@fortawesome/pro-light-svg-icons';
import Courses from '../../features/Courses';
import AnnouncementContainer from '../../ui/AnnouncementContainer';
import { MainGridWrapper, MainGrid, MainGridCol, SecondGridWrapper } from '../../ui/PageGrid';
import PlannerItems from '../../features/PlannerItems';
import AcademicCalendar from '../../features/AcademicCalendar';
import PageTitle, { Title } from '../../ui/PageTitle';
import AcademicOverview from '../../features/AcademicOverview';
import ResourcesCard from '../../features/ResourcesCard';
import { AcademicSubNav } from './AcademicsSubNav';

const AcademicsDashboard = () => {
  return (
    <>
      <MainGridWrapper>
        <PageTitle title="Academics" />
        <AcademicSubNav />
        <MainGrid data-testid="academics-dashboard">
          <MainGridCol>
            <AcademicOverview />
            <PlannerItems />
            <Courses />
          </MainGridCol>
          <MainGridCol>
            <ResourcesCard categ="academic" icon={faGraduationCap} />
            <AcademicCalendar />
          </MainGridCol>
        </MainGrid>
      </MainGridWrapper>
      <SecondGridWrapper>
        <Title as="h2">Announcements and Events</Title>
        <AnnouncementContainer className="col-span-2" type="academic" />
      </SecondGridWrapper>
    </>
  );
};

export default AcademicsDashboard;
