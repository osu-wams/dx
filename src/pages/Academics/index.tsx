import React from 'react';
import { Router, RouteComponentProps } from '@reach/router';
import PastCourses from './PastCourses';
import AcademicsDashboard from './AcademicsDashboard';

const RouterPage = (props: { pageComponent: JSX.Element } & RouteComponentProps) =>
  props.pageComponent;

const Academics = () => {
  return (
    <Router className="router-styles">
      <RouterPage pageComponent={<AcademicsDashboard />} path="/" />
      <RouterPage pageComponent={<PastCourses />} path="past-courses" />
    </Router>
  );
};

export default Academics;
