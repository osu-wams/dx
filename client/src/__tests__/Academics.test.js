import React from 'react';
import { render } from '../componentTestUtils';
import Academics from '../components/pages/Academics';

test('renders', () => {
  const { container } = render(<Academics />);
  expect(container.firstChild).toMatchSnapshot();
});
