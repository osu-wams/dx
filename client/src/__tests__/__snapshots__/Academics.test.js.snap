// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.c0 {
  background-color: #ffffff;
  border-radius: 8px;
  box-shadow: 0px 1px 3px 0px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 2px 1px -1px rgba(0,0,0,0.12),0px -8px 0px 0px #006a8e;
  overflow: hidden;
  -webkit-transition: box-shadow 0.1s linear;
  transition: box-shadow 0.1s linear;
  margin-top: 8px;
  margin-bottom: 24px;
}

.c1 {
  border: 0;
  background: transparent;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 16px;
  min-height: 68px;
  line-height: 68px;
}

.c1:hover {
  cursor: initial;
}

.c1 svg {
  margin-right: 16px;
}

.c1.cardheader-clickable {
  cursor: pointer;
}

.c2 {
  font-size: 24px;
  font-family: 'Stratum2',sans-serif;
  font-weight: bold;
}

.c4 {
  font-family: 'Open Sans',sans-serif;
  font-size: 16px;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  overflow: hidden;
  max-height: 0;
  padding: 0;
  padding-top: 0;
}

.c3 {
  color: #006a8e;
}

.c6 {
  height: 22px;
  min-width: 22px;
  line-height: 22px;
  text-align: center;
  font-weight: 600;
  background-color: #006a8e;
  color: #ffffff;
  border-radius: 2rem;
  font-size: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 8px;
  margin-right: 8px;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  color: #252525;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0;
}

.c8 li {
  list-style-type: none;
  border-top: 1px solid #e0e0e0;
}

.c8 li:last-child {
  border-bottom: 1px solid #e0e0e0;
}

.c8 li button {
  width: 100%;
  background: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 16px;
  font-size: 1.6rem;
  border: none;
}

.c8 li button div {
  color: #757575;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  text-align: left;
}

.c8 li button svg {
  margin-left: auto;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #e0e0e0;
}

@media screen and (min-width:768px) {
  .c0 {
    margin-top: 8px;
    margin-bottom: 0;
  }
}

<div
  data-testid="academics-page"
>
  <div
    style="border: 0px; height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; position: absolute;"
  >
    <h2>
      Academics
    </h2>
  </div>
  <div
    class="c0"
    color="stratosphere"
    elevation="1"
  >
    <button
      class="cardheader-clickable c1"
    >
      <div
        class="c2"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chalkboard-teacher fa-w-20 c3"
          color="stratosphere"
          data-icon="chalkboard-teacher"
          data-prefix="fas"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M208 352c-2.39 0-4.78.35-7.06 1.09C187.98 357.3 174.35 360 160 360c-14.35 0-27.98-2.7-40.95-6.91-2.28-.74-4.66-1.09-7.05-1.09C49.94 352-.33 402.48 0 464.62.14 490.88 21.73 512 48 512h224c26.27 0 47.86-21.12 48-47.38.33-62.14-49.94-112.62-112-112.62zm-48-32c53.02 0 96-42.98 96-96s-42.98-96-96-96-96 42.98-96 96 42.98 96 96 96zM592 0H208c-26.47 0-48 22.25-48 49.59V96c23.42 0 45.1 6.78 64 17.8V64h352v288h-64v-64H384v64h-76.24c19.1 16.69 33.12 38.73 39.69 64H592c26.47 0 48-22.25 48-49.59V49.59C640 22.25 618.47 0 592 0z"
            fill="currentColor"
          />
        </svg>
        Courses
      </div>
      <div
        class="c4"
      >
        <span
          class="c5"
        >
          <span
            class="c6"
          >
            8
          </span>
          Credits
        </span>
      </div>
    </button>
    <div
      aria-expanded="false"
      class="c7"
    >
      Fall 2016
      <ul
        class="c8"
      >
        <li>
          <button
            type="button"
          >
            <span>
              Introduction To Computer Networks
              <div>
                CS
                 
                372
              </div>
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-chevron-right fa-w-10 "
              data-icon="chevron-right"
              data-prefix="fas"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
                fill="currentColor"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            type="button"
          >
            <span>
              Operating Systems I
              <div>
                CS
                 
                344
              </div>
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-chevron-right fa-w-10 "
              data-icon="chevron-right"
              data-prefix="fas"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
                fill="currentColor"
              />
            </svg>
          </button>
        </li>
      </ul>
    </div>
  </div>
  <p>
    Thisis: 
    Api
  </p>
</div>
`;
